src/libclipper/include/clipper/selection_policies.hpp

```diff
@@ -8,324 +8,123 @@
-                        long query_id);
-
-  static BanditPolicyState process_feedback(BanditPolicyState state,
-                                            Feedback feedback,
-                                            std::vector<Output> predictions);
-
-  static std::string serialize_state(BanditPolicyState state);
-
-  static BanditPolicyState deserialize_state(const std::string& bytes);
-
-  static std::string state_debug_string(const BanditPolicyState& state);
+  static Output deserialize(std::string serialized_state);
 };
 
-class EpsilonGreedyPolicy : public SelectionPolicy<Exp4Policy> {
-  // Epsilon Greedy
-  // Select: epsilon chance randomly select,
-  //         (1-epsilon) change select model with the highest expected reward
-  // Update: update individual model expected reward
-
+class DefaultOutputSelectionPolicy : public SelectionPolicy {
  public:
-  EpsilonGreedyPolicy() = delete;
-  ~EpsilonGreedyPolicy() = delete;
-  typedef BanditPolicyState state_type;
-
-  constexpr static double epsilon = 0.1;  // Random Selection Chance
-
-  static BanditPolicyState initialize(
-      const std::vector<VersionedModelId>& candidate_models);
-
-  static BanditPolicyState add_models(
-      BanditPolicyState state, const std::vector<VersionedModelId>& new_models);
-
-  static long hash_models(
-      const std::vector<VersionedModelId>& /*candidate_models*/) {
-    return 0;
-  };
-
-  static std::vector<PredictTask> select_predict_tasks(BanditPolicyState& state,
-                                                       Query query,
-                                                       long query_id);
-
-  static Output combine_predictions(BanditPolicyState state, Query query,
-                                    std::vector<Output> predictions);
-
-  static std::pair<std::vector<PredictTask>, std::vector<FeedbackTask>>
-  select_feedback_tasks(BanditPolicyState& state, FeedbackQuery feedback,
-                        long query_id);
-
-  static BanditPolicyState process_feedback(BanditPolicyState state,
-                                            Feedback feedback,
-                                            std::vector<Output> predictions);
-
-  static std::string serialize_state(BanditPolicyState state);
-
-  static BanditPolicyState deserialize_state(const std::string& bytes);
-
-  static std::string state_debug_string(const BanditPolicyState& state);
-
- private:
-  static VersionedModelId select(BanditPolicyState& state);
+  DefaultOutputSelectionPolicy() = default;
+  DefaultOutputSelectionPolicy(const DefaultOutputSelectionPolicy&) = default;
+  DefaultOutputSelectionPolicy& operator=(const DefaultOutputSelectionPolicy&) =
+      default;
+  DefaultOutputSelectionPolicy(DefaultOutputSelectionPolicy&&) = default;
+  DefaultOutputSelectionPolicy& operator=(DefaultOutputSelectionPolicy&&) =
+      default;
+  ~DefaultOutputSelectionPolicy() = default;
+  std::shared_ptr<SelectionState> init_state(Output default_output) const;
```

Comment:
The problem with defining a virtual init_state method is that it's likely the arguments to it will vary based on the selection policy. So I'm not sure what the function signature would be. For the 0.1 release, we only have one selection policy (the DefaultOutputSelectionPolicy). init_state is called each time a new application is created, which is the correct behavior for _this_ selection policy. When we expand our functionality in 0.2 to support more sophisticated types of selection policies, we will need to revisit this. But I already knew we would need to refine our selection policy API anyway. [CLIPPER-99](https://clipper.atlassian.net/browse/CLIPPER-99) will track this effort.